test/test.js

```diff
@@ -0,0 +1,145 @@
+const { describe, it } = require('mocha')
+
+const assert = require('assert')
+
+const registry = require('prismarine-registry')('1.8')
+
+const windows = require('..')('1.8')
+const Item = require('prismarine-item')('1.8')
+
+const chest = windows.createWindow(1, 'minecraft:chest', 'chest', 27)
+const inv = windows.createWindow(2, 'minecraft:inventory', 'inventory')
+
+const createClick = (mode, mouseButton, slot, useInv = false) => ({
+  slot,
+  mouseButton,
+  mode,
+  windowId: useInv ? inv.id : chest.id,
+  item: slot === -999 ? null : useInv ? inv.slots[slot] : chest.slots[slot]
+})
+
+const stoneId = registry.itemsByName.stone.id
+const grassId = registry.itemsByName.grass.id
+
+chest.updateSlot(0, new Item(stoneId, 64))
+chest.updateSlot(1, new Item(stoneId, 1))
+chest.updateSlot(3, new Item(stoneId, 63))
+chest.updateSlot(5, new Item(grassId, 3))
+chest.updateSlot(chest.inventoryEnd - 1, new Item(stoneId, 44))
+chest.updateSlot(chest.inventoryEnd - 2, new Item(stoneId, 44))
+chest.updateSlot(chest.inventoryEnd - 3, new Item(stoneId, 44))
+
+inv.updateSlot(inv.inventoryEnd - 1, new Item(registry.itemsByName.leather_boots.id, 1))
+inv.updateSlot(inv.inventoryEnd - 2, new Item(stoneId, 1))
+
+describe('mode 0 | normal click', () => {
+  describe('mouseButton 0', () => {
+    it('pickup item', () => {
+      chest.acceptClick(createClick(0, 0, 0))
+      assert(!chest.slots[0] && chest.selectedItem?.count === 64 && chest.selectedItem.type === 1)
```

Comment:
Don't put multiple assertions into one assert. This will make a test unreadable if it does fail. The assertion is also missing a meaningful comment of what is being asserted. For instance this line should be changed to something like this.
```js
assert(!chest.slots[0], 'Slot 0 should be empty')
assert(chest.selectedItem?.count === 64 && chest.selectedItem?.type === 1, 'Selected item in chest should be stone (??) and have count 64')
```